README.md

```diff
@@ -1,2 +1,319 @@
+```
+
+More complex toasts,
+that don't fit the above content model,
+would require more custom handling on the part of the developer:
+
+```html
+<std-toast>
+  <p>Hello world!</p>
+  <p>A second paragraph?!</p>
+
+  <button>Action 1</button>
+  <button>Action 2</button>
+</std-toast>
+```
+
+Such an unusual toast would still integrate with other toasts in terms of stacking and positioning behavior,
+and some of the default styles that are inherited may be useful.
+But the page author will need to handle the action clicks themselves (instead of using the `"actionclick"` event),
+and will need to add additional styling to handle the extra contents.
+
+TODO: when we have a prototype, link to/show an example of this in action.
+
+#### Methods
+
+The `.show(options)` function shows a toast element,
+by toggling its `open=""` attribute to true.
+The `options` include:
+
+- `duration`: how long to show the toast, in milliseconds. Defaults to ???
+- `multiple`: TODO
+- `newestOnTop`: TODO
+
+TODO: how do `multiple` and `newestOnTop` work?
+Should those be a global setting?
+Per container?
+How do we deal with different toasts having different values?
+
+#### Events
+
+A `<std-toast>` element can fire the following events:
+
+- `"open"`: the toast was shown
+- `"close"`: the toast was closed, either explicitly by the user, or via the timeout.
+  (Note: if animations were applied, the toast may not be entirely invisible at the time this event fires)
+    - TODO: should we consider separate events for the start and end of any close animation?
+      This seems hard to do correctly if the user customizes the animation, though.
+- `"actionclick"`: the toast's call-to-action button or link was clicked, if one exists TODO
+
+### `showToast(message, options)`
+
+The `"std:elements/toast"` module also exports a convenience function,
+`showToast()`,
+which allows creating and showing toasts entirely from JavaScript.
+Behind the scenes,
+`showToast()` creates a `<std-toast>`,
+sets it up using the given message and options,
+inserts it as the last child of `<body>`,
+and then adds the `open=""` attribute,
+to make the toast visible.
+Finally,
+it returns the created `<std-toast>` element,
+allowing further manipulation by script.
+
+TODO: will `showToast()` support putting the toast into a specified container,
+or only into `body`?
+
+`message` is a string that will be inserted as a text node
+(TODO: or as a `<p>` element?) into the created `<std-toast>`.
+
+TODO: can `message` contain HTML?
+
+`options` allows configuring both the attributes of the `<std-toast>`,
+and the options for this particular showing of the toast.
+Thus, the possible options are:
+
+- `theme`, like the attribute
+- `position`, like the attribute
+- `closeButton`, like the attribute
+- `duration`, like the `show()` option
+- `multiple`, like the `show()` option
+- `newestOnTop`, like the `show()` option
+
+TODO: how do we deal with actions with this API? Ideas:
```

Comment:
By "deal with actions", you mean how does the author specify actions? Do we need to provide this? It seems pretty easy to provide your own buttons or to wrap this and provide actions.

If we do, can just take a list of `[['Button text', callback], ...], that will work fine with a `message` toast (maybe expose a numbered CSS `part` for each one and a `part` that selects them all). If the author is doing a HTML toast then they can easily do anything they want for buttons and I wonder what value we would be adding by including a way of doing actions in that case.

Reply:
Right, this is about how the author specifies actions for the JavaScript API. I agree that for a HTML toast this is not an issue.

The author could always do it themselves, but if the point of the JS API is to be easy to use like other libraries, I think

```js
const toast = showToast("message");
const button = document.createElement("button");
button.textContent = "Click me";
button.onclick = doSomething;
toast.append(button);
```

is not super user-friendly.

Your proposal is, I think,

```js
const toast = showToast("message", {
  action: ['Click me', doSomething]
});
```

which is one reasonable nicer API shape. But there are other possibilities too, and we were thinking of leaving this as a TODO and discussing the possibilities in an issue. (E.g., what if you want an `<a>` instead of a `<button>`?)

Reply:
It seems showToast is a convenience wrapper around std-toast and that's where the action is useful. But std-toast needs actionclick to service showToast. Am I reading that right? That layering seems a bit strange.

This feels like the sort of thing that should be an open issue that we inform primarily with partner feedback.

Reply:
Yeah. Issue sounds good.

Reply:
#11